### ISSUE TITLE
Why normalize the intrinsics and extrinsics in the covert_dtu.py? 

### ISSUE THREAD

**Issue by Miaosheng1** [donydchen/mvsplat#28]
For the DTU dataset, I noticed your code has normalized the camera intrinsics in the convert.py? As a result, the pose and intrinsics in the code are hard to understand and it is not convenient to use this code for 3rd dataset (Waymo or Mipnerf360).

**Comment by donydchen** [comment]
Hi @Miaosheng1, thanks for your interest in our work.

We mainly normalised the DTU intrinsic to align with our default RE10K data loader, which assumes a normalised intrinsic as input. Besides, since we use the camera parameters in different resolutions, e.g., image and feature resolutions (for building the feature cost volume), it would be more convenient to normalise them from the input and scale them back to a specific resolution when needed.

Normalising the intrinsic mainly involves dividing it by the initial height and width. Suppose the original intrinsic is

```python
K = [[width * focal_length_x,             0,            width * principal_point_x],
     [          0           , height * focal_length_y, height * principal_point_y],
     [          0           ,             0,                       1             ]]
```

Then, you just need to divide the first row by width and the second by height, and that's it. The remaining parts of the `convert_dtu.py` mainly aim to align the cropped data with the initial camera parameters, which is unnecessary for other datasets.

For the camera extrinsic, MVSplat uses the OpenCV-style camera-to-world matrices. So, basically, +Z is camera look, +X is camera right, and -Y is camera up. 

Let me know if you have encountered any other difficulties. Cheers.

**Comment by Miaosheng1** [comment]
Thank you for your clear response!   I also noticed that the **scale** parameters are set to 200 in convert.py, this means that the translation is divided 200 in your code.  How do you set the hyperparameter and what does it mean? Besides, as you normalize the intrinsics, the input image is cropped in this code  **function:  apply_crop_shim**.  So my question is what if i do not crop the image(means **cx !=  image/width**) in dtu and do not normalize the intrinsics at the same time, dose it work?

**Comment by Miaosheng1** [comment]
Thank you for your suggestion. In your provided DTU dataset,  the image size is **(512×640)**. In your code, the image was cropped (**mainly in function: apply_crop_shim**), and the processed image size is (256×256). This process definitely lead the principal point locate the center of the image.  I think this processed image(256×256) corresponding to the normalized intrinsic you mentioned.  When i change the image size (256×256) -> (512×640) in **dtu.yaml**, it leads some ghost artifacts, see blew. Should I change the camera intrinsic  when i want to render the image with origin size (512×640). 
![000032](https://github.com/donydchen/mvsplat/assets/111415805/58558e4e-6048-4895-a5be-c02f9f5969ee)

**Comment by donydchen** [comment]
Hi @Miaosheng1, in our code, we assume the image shapes are even (see [here](https://github.com/donydchen/mvsplat/blob/378ff818c0151719bbc052ac2797a2c769766320/src/dataset/shims/patch_shim.py#L8)) and mainly perform the center crop operation, which is symmetric and should not change the _normalized_ principal point. Since you use the DTU original shape and do not perform any operation like asymmetric cropping, the normalized principal point is still at `(0.5, 0.5)`. The convert script only normalises the original DTU-provided camera intrinsic, which is not tied to the resolution in our MVSplat data loader.

In fact, the Gaussian Splatting rasterizer we used _only_ supports the principal point at the image center, so consider this when applying any related operations.

Additionally, I think the results you get look reasonably good. Since the model is trained on the `256x256` real estate dataset, its performance might be suboptimal when applied to other resolutions and/or data types. To get better performances, you might consider fine-tuning the model on the DTU training set with its original shape.

**Comment by Miaosheng1** [comment]
Thanks

**Comment by donydchen** [comment]
Hi @ShunyuanZheng, you are correct. The original principal point of DTU is _slightly_ off-center; we force it to be in the center for simplicity.

Take the scene 'scan1_train' for an example, by going through the preprocessing, its intrinsic (at [here](https://github.com/donydchen/mvsplat/blob/378ff818c0151719bbc052ac2797a2c769766320/src/scripts/convert_dtu.py#L136)) is 

```python
intr = [[1446.165, 0.0,     331.602 ], 
        [0.0,      1441.59, 265.5345], 
        [0.0,      0.0,     1.0.    ]]
```

Since its image shape is (512, 640), the correct normalized principal point should be around **(0.5181, 0.5186)**, which is slightly off-center. Of course, one can apply cropping to shift the principal point to the exact center. Still, I think that forcing it to be in the center is within reasonable variation and easy to understand for quick testing.

**Comment by ShunyuanZheng** [comment]
> Hi @ShunyuanZheng, you are correct. The original principal point of DTU is _slightly_ off-center; we force it to be in the center for simplicity.
> 
> Take the scene 'scan1_train' for an example, by going through the preprocessing, its intrinsic (at [here](https://github.com/donydchen/mvsplat/blob/378ff818c0151719bbc052ac2797a2c769766320/src/scripts/convert_dtu.py#L136)) is
> 
> ```python
> intr = [[1446.165, 0.0,     331.602 ], 
>         [0.0,      1441.59, 265.5345], 
>         [0.0,      0.0,     1.0.    ]]
> ```
> 
> Since its image shape is (512, 640), the correct normalized principal point should be around **(0.5181, 0.5186)**, which is slightly off-center. Of course, one can apply cropping to shift the principal point to the exact center. Still, I think that forcing it to be in the center is within reasonable variation and easy to understand for quick testing.

I see. Thanks for your quick reply!

**Comment by FudongGe** [comment]
Hi, if only resizing without cropping, is it necessary to adjust the intrinsics (already been normalized)? In other words, what should be the intrinsics adjustments for center crop and resize, respectively (if intrinsics have been normalized)？

**Comment by FudongGe** [comment]
Hi, if only resizing without cropping, is it necessary to adjust the intrinsics (already been normalized)? In other words, what should be the intrinsics adjustments for center crop and resize, respectively (if intrinsics have been normalized)？

Specifically, why is there no scaling of intrinsics in the rescale function？
![image](https://github.com/user-attachments/assets/726bea41-d2ab-462a-ba51-7b6f41853cd7)
